File: Data/Entities/TeacherSubject.cs
using System;

namespace Magnifinance.Data.Entities
{
    public class TeacherSubject
    {
        public int SubjectTeacherId { get; set; }
        public string Note { get; set; }

        public int TeacherId { get; set; }
        public int SubjectId { get; set; }

        public Teacher Teacher { get; set; }
        public Subject Subject { get; set; }
    }
}

File: client/src/app/shared/TeacherSubject.ts
import { ISubject } from "./Subject";
import { ITeacher } from "./Teacher";

export class ITeacherSubject {
    subjectTeacherId: number;
    note: string;
    teacherId: number;
    subjectId: number;
    teacher: ITeacher[];
    subject: ISubject[]
}